src/ol/style/RegularShape.js

```diff
@@ -505,6 +505,9 @@ class RegularShape extends ImageStyle {
     this.renderOptions_ = this.createRenderOptions();
     const size = this.renderOptions_.size;
     this.canvas_ = {};
+    if (this.size_ && (this.size_[0] != size || this.size_[1] != size)) {
+      this.hitDetectionCanvas_ = null;
+    }
```

Comment:
We don't currently properly hit detect circles after a call to `circle.setRadius()`.  I pulled this change out in a separate pull request with tests: #15143.